src/prisma/schema.prisma

```diff
@@ -0,0 +1,93 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id                     String         @id @default(cuid())
+  email                  String?        @unique
+  address                String         @unique
+  listings               Listing[]
+  auctions               Auction[]
+  initiatedListingOffers ListingOffer[] @relation("initiatedListingOffers")
+  receievedListingOffers ListingOffer[] @relation("receivedListingOffers")
+  initiatedAuctionBids   AuctionBid[]   @relation("initiatedAuctionBids")
+  receivedAuctionBids    AuctionBid[]   @relation("receivedAuctionBids")
+}
+
+model Listing {
+  id          String         @id @default(cuid())
+  creatorAddr String
+  tokenId     String         @unique
+  price       Decimal
+  createdAt   DateTime       @default(now())
+  updatedAt   DateTime       @updatedAt()
+  offers      ListingOffer[]
+  nft         NFT            @relation(fields: [tokenId], references: [tokenId], onDelete: Cascade)
+  creator     User           @relation(fields: [creatorAddr], references: [address], onDelete: Cascade)
+}
+
+model ListingOffer {
+  id          String   @id @default(cuid())
+  offerorAddr String
+  offereeAddr String
+  amount      Decimal
+  listingId   String
+  createdAt   DateTime @default(now())
+  offeror     User     @relation(name: "initiatedListingOffers", fields: [offerorAddr], references: [address], onDelete: Cascade)
+  offeree     User     @relation(name: "receivedListingOffers", fields: [offereeAddr], references: [address], onDelete: Cascade)
+  listing     Listing  @relation(fields: [listingId], references: [id], onDelete: Cascade)
+}
+
+model Auction {
+  id          String       @id @default(cuid())
+  creatorAddr String
+  tokenId     String       @unique
+  createdAt   DateTime     @default(now())
+  duration    Decimal
+  minBid      Decimal
+  updatedAt   DateTime     @updatedAt()
+  bids        AuctionBid[]
+  nft         NFT          @relation(fields: [tokenId], references: [tokenId], onDelete: Cascade)
+  creator     User         @relation(fields: [creatorAddr], references: [address], onDelete: Cascade)
+}
+
+model AuctionBid {
+  id           String   @id @default(cuid())
+  bidderAddr   String
+  receiverAddr String
+  amount       Decimal
+  auctionId    String
+  createdAt    DateTime @default(now())
```

Comment:
nitpick: don't you think all models should have `createdAt` & `updatedAt`?

Reply:
I think only auction and listing offer requires and updatedAt field presently